core/server/common/src/main/java/alluxio/cli/validation/HdfsValidationTask.java

```diff
@@ -0,0 +1,135 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.cli.validation;
+
+import alluxio.Configuration;
+import alluxio.PropertyKey;
+
+import org.apache.commons.cli.Option;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Abstract class for validating HDFS-related configurations.
+ */
+public class HdfsValidationTask extends AbstractValidationTask {
+  /** Name of the environment variable to store the path to Hadoop config directory. */
+  protected static final String HADOOP_CONF_DIR_ENV_VAR = "HADOOP_CONF_DIR";
+
+  protected static final Option HADOOP_CONF_DIR_OPTION =
+      Option.builder("hadoopConfDir").required(false).hasArg(true)
+      .desc("path to server-side hadoop conf dir").build();
+
+  /**
+   * Constructor of {@link HdfsValidationTask}.
+   */
+  public HdfsValidationTask() {}
+
+  /**
+   * {@inheritDoc}
+   */
+  @Override
+  public List<Option> getOptionList() {
+    List<Option> opts = new ArrayList<>();
+    opts.add(HADOOP_CONF_DIR_OPTION);
+    return opts;
+  }
+
+  @Override
+  public boolean validate(Map<String, String> optionsMap) {
+    if (!validateHdfsSettingParity(optionsMap)) {
+      System.err.format("Hdfs setting do not match.");
+      return false;
+    }
+    return true;
+  }
+
+  private boolean validateHdfsSettingParity(Map<String, String> optionsMap) {
+    String serverHadoopConfDirPath;
+    if (optionsMap.containsKey(HADOOP_CONF_DIR_OPTION.getOpt())) {
+      serverHadoopConfDirPath = optionsMap.get(HADOOP_CONF_DIR_OPTION.getOpt());
+    } else {
+      serverHadoopConfDirPath = System.getenv(HADOOP_CONF_DIR_ENV_VAR);
+    }
+    if (serverHadoopConfDirPath == null) {
+      System.err.println("Path to server-side hadoop configuration unspecified,"
+          + " skipping validation for HDFS properties.");
+      return true;
+    }
+    String serverCoreSiteFilePath = serverHadoopConfDirPath + "/core-site.xml";
+    String serverHdfsSiteFilePath = serverHadoopConfDirPath + "/hdfs-site.xml";
+
+    // If Configuration does not contain the key, then a {@link RuntimeException} will be thrown
+    // before calling the {@link String#split} method.
+    String[] clientHaoopConfFilePaths =
+        Configuration.get(PropertyKey.UNDERFS_HDFS_CONFIGURATION).split(":");
+    String clientCoreSiteFilePath = clientHaoopConfFilePaths[0];
```

Comment:
the order of core/hdfs-site.xml in the config parameter is not guaranteed, i.e, the user could specify hdfs-site first

Reply:
instead of parsing the files and comparing files individually, why not 
```
Configuration hdfsConf = new Configuration();
for (String path : conf.getValue(PropertyKey.UNDERFS_HDFS_CONFIGURATION).split(":")) {
  hdfsConf.addResource(new Path(path));
}
```

and then use `hdfsConf.iterator()` to iterate the keys

Reply:
This change requires adding Hadoop and HDFS dependencies to`core/server/common/pom.xml`. Since we support multiple Hadoop versions, we need to consider them all. In the current `pom.xml`, we don't have dependency on Hadoop/HDFS in this component. Since this is the core component, I feel we should discuss this.

Reply:
We discussed offline, and this may require us to re-arrange our entire validation-related code. Since this is not likely my recent priority, I suggest that we deal with this in a separate PR.